plenum/test/view_change/test_master_primary_different_from_previous.py

```diff
@@ -0,0 +1,81 @@
+import types
+
+from plenum.test.helper import checkViewNoForNodes, \
+    sendReqsToNodesAndVerifySuffReplies, countDiscarded
+from plenum.test.malicious_behaviors_node import slow_primary
+from plenum.test.test_node import getPrimaryReplica, ensureElectionsDone
+from plenum.test.pool_transactions.conftest import clientAndWallet1, client1, \
+    wallet1, client1Connected, looper
+from plenum.test.view_change.helper import chk_view_change
+from stp_core.loop.eventually import eventually
+
+
+def test_master_primary_different_from_previous1(txnPoolNodeSet,
+                                                 looper, client1,
+                                                 wallet1, client1Connected):
+    """
+    After a view change, primary must be different from previous primary for
+    master instance, it does not matter for other instance. The primary is
+    benign and does not vote for itself.
+    """
+    old_view_no = checkViewNoForNodes(txnPoolNodeSet)
+    pr = slow_primary(txnPoolNodeSet, 0, delay=10)
+    old_pr_node_name = pr.node.name
+
+    # View change happens
+    looper.run(eventually(chk_view_change, txnPoolNodeSet, old_view_no + 1,
+                          wallet1, client1, retryWait=1, timeout=60))
+
+    # Elections done
+    ensureElectionsDone(looper=looper, nodes=txnPoolNodeSet)
+    # New primary is not same as old primary
+    assert getPrimaryReplica(txnPoolNodeSet, 0).node.name != old_pr_node_name
+
+    pr.outBoxTestStasher.resetDelays()
+
+    # The new primary can still process requests
+    sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, 5)
+
+
+def test_master_primary_different_from_previous2(txnPoolNodeSet,
```

Comment:
Renaming the test will make the test name too big, the comment describes what the test does